### PR TITLE
feat(Header): use InternalLink for breadcrumb items

### PR DESCRIPTION
**Pull request by NaiSidera**

### CODE REVIEW

File: src/containers/Header/Header.tsx [hunk 1]
```diff
@@ -80,15 +81,22 @@ function Header({mainPage}: HeaderProps) {
...
                                     {text}
                                 </span>
+                            ) : (
+                                text
+                            );
+
+                            return (
+                                <InternalLink to={isCurrent ? undefined : link}>
+                                    {content}
+                                </InternalLink>
```
Comment (artemmufazalov) [inline src/containers/Header/Header.tsx:99]: Can we keep old styles, but with new behaviour?

Styles should be like this:
Screenshot 2024-06-13 at 16 59 35

But not they are:
Screenshot 2024-06-13 at 16 59 40

**Review by artemmufazalov — CHANGES_REQUESTED** [review]

File: src/containers/Tablets/Tablets.tsx [hunk 2]
```diff
@@ -34,7 +34,7 @@ const columns: DataTableColumn<TTabletStateInfo & {fqdn?: string}>[] = [
         render: ({row}) => {
             return (
                 <span>
-                    {row.Type} {row.Leader ? <Text color="secondary">leader</Text> : ''}
+                    {row.Type} {row.Leader ? '' : <Text color="secondary">follower</Text>}
```
Comment (Raubzeug) [inline src/containers/Tablets/Tablets.tsx:37]: Do you mind if we do it in separate PR?

File: src/containers/Header/breadcrumbs.tsx [hunk 3]
```diff
@@ -55,167 +61,141 @@ const getClusterBreadcrumbs = (
...
+        ? getClusterBreadcrumbs(options, query)
+        : getTenantBreadcrumbs(options, {...query, ...tenantQuery});
 
-    const text = nodeId
-        ? `${headerKeyset('breadcrumbs.node')} ${nodeId}`
-        : headerKeyset('breadcrumbs.node');
-    const link = nodeId ? getDefaultNodePath(nodeId, query) : undefined;
-    const icon = isStorageNode ? <StorageNodeIcon /> : <ComputeNodeIcon />;
+    let text = headerKeyset('breadcrumbs.node');
+    if (nodeId) text += ` ${nodeId}`;
```
Comment (Raubzeug) [inline src/containers/Header/breadcrumbs.tsx:90]: please add curly braces

File: src/containers/Header/breadcrumbs.tsx [hunk 4]
```diff
@@ -55,167 +61,141 @@ const getClusterBreadcrumbs = (
...
-        nodeId: nodeId,
+        nodeId,
     });
 
-    const text = pDiskId
-        ? `${headerKeyset('breadcrumbs.pDisk')} ${pDiskId}`
-        : headerKeyset('breadcrumbs.pDisk');
-    const link = pDiskId && nodeId ? getPDiskPagePath(pDiskId, nodeId, query) : undefined;
+    let text = headerKeyset('breadcrumbs.pDisk');
+    if (pDiskId) text += ` ${pDiskId}`;
```
Comment (Raubzeug) [inline src/containers/Header/breadcrumbs.tsx:111]: curly braces

File: src/containers/Header/breadcrumbs.tsx [hunk 5]
```diff
@@ -55,167 +61,141 @@ const getClusterBreadcrumbs = (
...
-            return [...rawBreadcrumbs, ...getTabletsBreadcrubms(options, query)];
-        }
-        case 'tablet': {
-            return [...rawBreadcrumbs, ...getTabletBreadcrubms(options, query)];
-        }
-        default: {
-            return rawBreadcrumbs;
-        }
-    }
+    if (!page) return rawBreadcrumbs;
```
Comment (Raubzeug) [inline src/containers/Header/breadcrumbs.tsx:193]: curly braces

File: src/containers/Header/breadcrumbs.tsx [hunk 6]
```diff
@@ -55,167 +61,141 @@ const getClusterBreadcrumbs = (
...
+
+    const getter = mapPageToGetter[page];
+
+    return [...rawBreadcrumbs, ...getter(options, query)].map((item) => ({
+        ...item,
+        action: () => {},
```
Comment (Raubzeug) [inline src/containers/Header/breadcrumbs.tsx:199]: let's return here raw breadcrumbs without action (cause you already add action in https://github.com/ydb-platform/ydb-embedded-ui/pull/913/files#diff-aa4640b8027696409b37009a150d396684a06d0dd3d7ec6c4abbeb6a74956480R51

**Review by Raubzeug — CHANGES_REQUESTED** [review]

File: src/containers/Header/Header.scss [hunk 7]
```diff
@@ -20,4 +20,16 @@
             margin-right: 3px;
         }
     }
+
+    .g-breadcrumbs {
```
[the remark this answers is not in the document]
Comment (NaiSidera) [reply src/containers/Header/Header.scss]: Done

File: src/containers/Header/Header.tsx [hunk 8]
```diff
@@ -1,8 +1,10 @@
 import React from 'react';
 
 import {Breadcrumbs} from '@gravity-ui/uikit';
+import _ from 'lodash';
```
[the remark this answers is not in the document]
Comment (NaiSidera) [reply src/containers/Header/Header.tsx]: Done

File: src/containers/Header/Header.tsx [hunk 9]
```diff
@@ -33,44 +35,32 @@ interface HeaderProps {
...
-            rawBreadcrumbs.push(mainPage);
-        }
-
-        if (clusterNameFinal) {
-            options = {
-                ...pageBreadcrumbsOptions,
-                clusterName: clusterNameFinal,
-            };
-        }
+        if (mainPage) rawBreadcrumbs.push(mainPage);
```
[the remark this answers is not in the document]
Comment (NaiSidera) [reply src/containers/Header/Header.tsx]: Done (all over)

File: src/containers/Tablets/Tablets.tsx (hunk 2 above)
Comment (NaiSidera) [reply src/containers/Tablets/Tablets.tsx]: Nope
Done!

File: src/containers/Header/breadcrumbs.tsx (hunk 6 above)
Comment (NaiSidera) [reply src/containers/Header/breadcrumbs.tsx]: I guess this placement is more appropriate but actually not a big deal
Done

File: src/containers/Header/Header.scss [hunk 10]
```diff
@@ -8,8 +8,17 @@
 
     padding: 0 20px 0 12px;
 
+    color: var(--g-color-text-primary);
     border-bottom: 1px solid var(--g-color-line-generic);
 
+    a {
```
Comment (Raubzeug) [inline src/containers/Header/Header.scss:14]: Lets not define styles for html-tags, it's bad practice, that may lead to collisions and unwanted side-effects. Please use custom classNames.

**Review by Raubzeug — CHANGES_REQUESTED** [review]

File: src/containers/Header/Header.scss (hunk 7 above)
[the remark this answers is not in the document]
Comment (NaiSidera) [reply src/containers/Header/Header.scss]: @Raubzeug btw, the solution is tricky enough
https://stackoverflow.com/questions/69883170/is-there-an-eslint-rule-to-prevent-assignment-in-a-single-line-if-statement

[the remark this answers is not in the document]
Comment (Raubzeug) [reply src/containers/Header/Header.scss]: It will be very [easy](https://eslint.org/docs/latest/rules/curly) when updated to eslint 9:)

File: src/containers/Header/Header.scss [hunk 11]
```diff
@@ -8,25 +8,24 @@
...
+            gap: 3px;
 
-    &__breadcrumb {
-        display: flex;
-        align-items: center;
+            color: var(--g-color-text-secondary);
 
-        &__icon {
-            display: flex;
+            &_icon {
```
[the remark this answers is not in the document]
Comment (Raubzeug) [reply src/containers/Header/Header.scss:21]: In our code, we try to avoid more than one level of nesting (except for modifiers). This means that entities should follow each other and not be enclosed.

**Review by Raubzeug — CHANGES_REQUESTED** [review]

**Review by Raubzeug — APPROVED** [review]
